### PR TITLE
propagating custom data to the callback handler

### PR DESCRIPTION
**Pull request by quaso**

### CODE REVIEW

**Comment by jonataswalker** [comment]
Hi @quaso! Would you mind to show a use case for your adding?

**Comment by quaso** [comment]
In "goto: work" item I have destination coordinates for work. So I can click anywhere on the map and coordinates stored in the item are used instead of the mouse click coordinates,
altuxln1uz

**Comment by jonataswalker** [comment]
Hi @quaso sorry this long delay but if have some more patience, please show me how would the application code with this changes. I guess I'm not being smart enough to get there.

**Comment by quaso** [comment]
https://github.com/quaso/mazda-enhanced-compass-routing/blob/export/MazdaRoutesEditor/src/main/resources/static/js/main.js

Look at function createMapContextMenu(), where the context menu is created and especially the for-loop at line 197, where the items with extra data are created.
And then at line 140, where the createFinishMarker function is defined. If additional data are present, then these are used instead of obj.coordinate.

File: src/html.js [hunk 1]
```diff
@@ -52,8 +52,12 @@
 
       ContextMenu.items[index] = {
         id: index,
-        callback: item.callback
+        callback: item.callback,
+        data:null
```
Comment (jonataswalker) [inline src/html.js]: This line could be: `data: item.data || null` and then remove next 3 lines.

File: src/internal.js [hunk 2]
```diff
@@ -83,7 +83,11 @@
             evt.preventDefault();
             var obj = {
               coordinate: this_.getCoordinateClicked()
+              data:null
```
Comment (jonataswalker) [inline src/internal.js]: Same here.

**Comment by jonataswalker** [comment]
I got it, good improvement, please see my inline code comments.

**Comment by quaso** [comment]
Fixed, thx.

**Comment by jonataswalker** [comment]
Merged. Many thanks @quaso.

**Pull request merged**